corehq/apps/locations/bulk_management.py

```diff
@@ -221,6 +220,17 @@ def needs_save(self):
             old_parent_code = ROOT_LOCATION_TYPE
         return old_parent_code != self.new_data.parent_code
 
+    def _metadata_needs_save(self):
+        # Save only for meaningful changes - not just to add empty fields
+        old_metadata, new_metadata = self.old_object.metadata, self.db_object.metadata
+        return (
+            # data is added or modified
+            any(old_metadata.get(k, '') != new_value
+                for k, new_value in new_metadata.items())
+            # data is removed
+            or set(old_metadata.keys()) - set(new_metadata.keys())
```

Comment:
can remove `.keys()` here to eliminate a stumbling block for Python 2 to 3 modernization tool

Reply:
I like `keys` in there for clarity - isn't this only an issue in a non-iterative context.

Reply:
technically yes, but I believe the python-modernize tool will flag this because it's not smart enough to figure out that type of contextual detail.

soft rant: I wish `dict.keys()` didn't exist. Iterating over a dict yields keys. "There should be one-- and preferably only one --obvious way to do it."

Reply:
> soft rant: I wish dict.keys() didn't exist. Iterating over a dict yields keys. "There should be one-- and preferably only one --obvious way to do it."

Haha, I feel the same, but with the opposite conclusion - `old_metadata.keys()` to me screams "we are extracting the keys from a dictionary", but iterating over the dictionary is much less obvious, "what's the data type of that thing again?".  IMO `.items()` seems like the more natural default for a dictionary, although I know python doesn't roll like that.

Now that you mention it though, I think this might be a more obvious implementation:
```python
any(k not in new_metadata for k in old_metadata.keys())
```

Reply:
In case you're interested in why iterating over a dict yields keys and not items, see [Why use dict.keys?](https://stackoverflow.com/a/17634737/10840), which references [PEP 234](https://www.python.org/dev/peps/pep-0234/) with the reasoning of the BDFL himself (find and read the section starting with "There has been a long discussion about whether").